src/flow/src/adapter/flownode_impl.rs

```diff
@@ -156,15 +173,14 @@ impl Flownode for FlowWorkerManager {
 
             let fetch_order = {
                 let ctx = self.node_context.read().await;
-                let table_col_names = ctx
-                    .table_repr
-                    .get_by_table_id(&table_id)
-                    .map(|r| r.1)
-                    .and_then(|id| ctx.schema.get(&id))
-                    .map(|desc| &desc.names)
-                    .context(UnexpectedSnafu {
-                        err_msg: format!("Table not found: {}", table_id),
-                    })?;
+
+                // TODO(discord9): also check schema version so that altered table can be reported
```

Comment:
Why not fix it right now?

Reply:
> Why not fix it right now?

Tried once, and realize it's a somewhere large change(need change metadata all the way, change proto and add compat tests etc) so want to put it in another PR